src/Compute/Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs

```diff
@@ -502,6 +511,58 @@ private AzureOperationResponse<VirtualMachine> UpdateVmEncryptionSettingsForMigr
             return setEncryptionEnabledFalse;
         }
 
+        private bool UpdateVmEncryptionIdentity()
+        {
+            bool updateVm = false;
+            var vmParameters = (this.ComputeClient.ComputeManagementClient.VirtualMachines.Get(
+                this.ResourceGroupName, this.VMName));
+
+            if (vmParameters.SecurityProfile == null)
+            {
+                vmParameters.SecurityProfile = new SecurityProfile();
+            }
+
+            if (vmParameters.SecurityProfile.EncryptionIdentity == null)
+            {
+                vmParameters.SecurityProfile.EncryptionIdentity = new EncryptionIdentity();
+            }
+
+            if (String.IsNullOrEmpty(vmParameters.SecurityProfile.EncryptionIdentity.UserAssignedIdentityResourceId) || !vmParameters.SecurityProfile.EncryptionIdentity.UserAssignedIdentityResourceId.Equals(this.EncryptionIdentity, StringComparison.OrdinalIgnoreCase))
+            {
+                vmParameters.SecurityProfile.EncryptionIdentity.UserAssignedIdentityResourceId = this.EncryptionIdentity;
+                updateVm = true;
+            }
+
+            if (updateVm)
+            {
+                // update VM
+                AzureOperationResponse<VirtualMachine> updateEncryptionIdentity = null;
+                updateEncryptionIdentity = this.ComputeClient.ComputeManagementClient.
+                    VirtualMachines.CreateOrUpdateWithHttpMessagesAsync(
+                        this.ResourceGroupName,
+                        vmParameters.Name,
+                        vmParameters).GetAwaiter().GetResult();
+
+                if (!updateEncryptionIdentity.Response.IsSuccessStatusCode)
+                {
+                    ThrowTerminatingError(new ErrorRecord(new ApplicationException(string.Format(CultureInfo.CurrentUICulture,
+                                                                                   Resources.EncryptionIdentityADEFailure,
+                                                                                    this.EncryptionIdentity, this.VMName,
+                                                                                    updateEncryptionIdentity.Response.Content.ReadAsStringAsync().GetAwaiter().GetResult())),
+                                                                                   "InvalidResult",
+                                                                                    ErrorCategory.InvalidResult,
+                                                                                    null));
+                }
+                else
+                {
+                    this.WriteObject(ComputeAutoMapperProfile.Mapper.Map<PSAzureOperationResponse>(updateEncryptionIdentity));
+                    return true;
+                }
+                return false;
+            }
+            return false;
```

Comment:
The function should return true if encryption identity is already assigned. Can you validate this?

Reply:
Will update it

Reply:
Done